src/Features/VisualBasic/Portable/ConvertToInterpolatedString/VisualBasicConvertRegularStringToInterpolatedStringRefactoringProvider.vb

```diff
@@ -0,0 +1,25 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Composition
+Imports System.Diagnostics.CodeAnalysis
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.ConvertToInterpolatedString
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.ConvertToInterpolatedString
+    <ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=PredefinedCodeRefactoringProviderNames.ExtractMethod), [Shared]>
+    Friend NotInheritable Class VisualBasicConvertRegularStringToInterpolatedStringRefactoringProvider
+        Inherits AbstractConvertRegularStringToInterpolatedStringRefactoringProvider(Of LiteralExpressionSyntax)
+
+        <ImportingConstructor>
+        <SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification:="Used in test code: https://github.com/dotnet/roslyn/issues/42814")>
+        Public Sub New()
+        End Sub
+
+        Protected Overrides Function IsAppropriateLiteralKind(literalExpression As LiteralExpressionSyntax) As Boolean
+            Return literalExpression.IsKind(SyntaxKind.StringLiteralExpression)
+        End Function
```

Comment:
if you use ISyntaxKinds.StringLiteralToken in the base class, you won't need this abstract/override in any of hte classes.  (the types can also become non-generic).

Reply:
So basically I could make `AbstractConvertRegularStringToInterpolatedStringRefactoringProvider` a concrete class that would handle both C# and VB refactoring, right?

Reply:
yup :)